pinot/net.py

```diff
@@ -109,24 +87,32 @@ def forward(self, g):
     def condition(self, g):
         """ Compute the output distribution.
         """
-        # get the parameters
-        theta = self.forward(g)
 
-        # put theta through noise model
-        theta = self.noise_model(theta)
+        if self.noise_model == 'normal-heteroschedastic':
+            mu, log_sigma = self.forward(g)
+            distribution = torch.distributions.normal.Normal(
+                    loc=mu,
+                    scale=torch.exp(log_sigma))
 
-        # parameterize the distribution according to the parameters
-        # and the distribution class
-        distribution = self.distribution_class(*theta)
+        elif self.noise_model == 'normal-homoschedastic':
+            mu, _ = self.forward(g)
 
-        return distribution
+            # initialize a `LOG_SIMGA` if there isn't one
+            if not hasattr(self, 'LOG_SIGMA'):
+                self.LOG_SIGMA = torch.tenosr(0.0)
```

Comment:
typo, do: ```torch.tensor```

And here now you can also bring in the measurement dimension, since you will have a log_sigma per dimension instead of a scalar.